src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -36,12 +36,23 @@ public class PackageArchiveReader : PackageReaderBase
         /// </summary>
         protected Stream ZipReadStream { get; set; }
 
+        public override bool RequiredRepoSign { get; }
+
+        public override bool PackageSignatureVerified { get; }
+
+        public override IEnumerable<IRepositoryCertInfo> RepositoryCertInfos { get; }
+
         /// <summary>
         /// Nupkg package reader
         /// </summary>
         /// <param name="stream">Nupkg data stream.</param>
-        public PackageArchiveReader(Stream stream)
-            : this(stream, false, DefaultFrameworkNameProvider.Instance, DefaultCompatibilityProvider.Instance)
+        /// <param name="requiredRepoSign">Set to true if the package is needed to be repository signed.</param>
+        public PackageArchiveReader(
+                Stream stream,
+                bool packageSignatureVerified = false,
```

Comment:
why package reader is taking `packageSignatureVerified` field? isn't this reader job to figure out that by going through package content?